src/accordion.ts

```diff
@@ -14,54 +15,37 @@ export const AccordionSummary = HTML.extend<HTMLElement>("MUI Accordion Summary"
     },
   });
 
-export const AccordionDetails = createInteractor<HTMLElement>("MUI Accordion Details")
-  .selector(".MuiAccordionDetails-root")
-  .filters({
-    id: (element) => element.id,
-    visible: { apply: isVisible, default: true },
-    className: (element) => element.className,
-    classList: (element) => Array.from(element.classList),
-  });
-
-export const AccordionActions = createInteractor<HTMLElement>("MUI Accordion Actions")
-  .selector(".MuiAccordionActions-root")
-  .filters({
-    id: (element) => element.id,
-    visible: { apply: isVisible, default: true },
-    className: (element) => element.className,
-    classList: (element) => Array.from(element.classList),
-  });
-
-export const Accordion = createInteractor<HTMLElement>("MUI Accordion")
+export const Accordion = HTML.extend<HTMLElement>("MUI Accordion")
   .selector(".MuiAccordion-root")
   .locator((element) => {
     const summary = getSummary(element);
     return isHTMLElement(summary) ? summary.getAttribute("aria-label") ?? summary.innerText : "";
   })
   .filters({
-    id: (element) => element.id,
-    className: (element) => element.className,
-    classList: (element) => Array.from(element.classList),
-    expanded: (element) => getSummary(element)?.getAttribute("aria-expanded") == "true",
+    expanded: isExpanded,
     disabled: {
       apply: (element) => getSummary(element)?.getAttribute("aria-disabled") == "true",
       default: false,
     },
   })
   .actions({
     expand: async (interactor) => {
-      try {
-        await interactor.is({ expanded: true, disabled: or(true, false) });
-      } catch (_) {
-        await interactor.find(AccordionSummary()).click();
-      }
+      let expanded = false;
+
+      await interactor.perform((element) => (expanded = isExpanded(element)));
+
+      if (expanded) return;
```

Comment:
Should we extract this into a `async getValue(interactor, getter)` helper?

Reply:
Because we have `setValue` in helpers, I thought what adding `getValue` would be a little confusing, so I called new helper `applyGetter`